invokeai/backend/model_manager/util/model_util.py

```diff
@@ -133,3 +133,16 @@ def _get_shape_1(key: str, tensor: torch.Tensor, checkpoint: Dict[str, torch.Ten
             break
 
     return lora_token_vector_length
+
+
+def convert_bundle_to_flux_transformer_checkpoint(
+    transformer_state_dict: dict[str, torch.Tensor],
+) -> dict[str, torch.Tensor]:
+    original_state_dict: dict[str, torch.Tensor] = {}
+    for k, v in transformer_state_dict.items():
+        if not k.startswith("model.diffusion_model"):
+            continue
+        if k.endswith("scale"):
+            v = v.to(dtype=torch.bfloat16)
```

Comment:
If I'm not mistaken, this will cause both the old float8 and new float16 tensor to be held in memory simultaneously (`transformer_state_dict` still holds a reference to the float8 tensor, so it can't be gc'd). I'm not sure how much total memory is used by scale parameters, but it's probably best to do this conversion in-place to keep memory usage as low as possible during load.

Reply:
Great callout, updated to remove this excess memory usage